frontend/src/App.jsx

```diff
@@ -25,26 +25,33 @@ import API from '@/API';
 import UserContext from '@/context/user';
 import AlertContext from '@/context/alert';
 import BiolinkContext from '@/context/biolink';
+import ConceptsContext from '@/context/concepts';
+
+import biolinkUtils from '@/utils/biolink';
 
 export default function App() {
   const [user, setUser] = useState(null);
+  const [biolink, setBiolink] = useState(null);
+  const [alert, setAlert] = useState({});
+  const [concepts, setConcepts] = useState([]);
```

Comment:
This is duplicated state, where we could just use
```javascript
const concepts = Object.keys(biolinkUtils.getValidConcepts(biolink));
```

I think we should try to never duplicate state if at all possible.

Reply:
See my other comment.

Reply:
Even if we keep the ConceptsContext, I think we should remove the extra state here. We can change the concepts to the following to not recompute on re-renders:

```javascript
const concepts = useMemo( Object.keys(biolinkUtils.getValidConcepts(biolink)), [biolink] );
```

Reply:
We could do it this way, but we'd also have to add some extra code to the getValidConcepts function to check if biolink is null (because it will be on the initial render). I'm really ok with the extra state how it is. It's not a complete duplicate of state, as we're converting the biolink classes to a trimmed down list.